packages/admin-app/src/components/ElectionSetupWizard/Steps/JointKeyUploadStep.tsx

```diff
@@ -0,0 +1,145 @@
+        paddingBottom: theme.spacing(4),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    spaced: {
+        marginBottom: theme.spacing(2),
+    },
+    buttonProgress: {
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        marginTop: -12,
+        marginLeft: -12,
+    },
+    error: {
+        color: 'red',
+        marginBottom: theme.spacing(2),
+    },
+}));
+
+export interface JointKeyUploadStepProps {
+    onNext: () => void;
+    onChanged: (newSubmitElectionRequest: SubmitElectionRequest) => void;
+}
+
+/**
+ * Joint Key Select Step for Election Setup
+ */
+const JointKeyUploadStep: React.FC<JointKeyUploadStepProps> = ({ onNext, onChanged }) => {
+    const [uploading, setUploading] = useState(false);
+    const [error, setError] = useState<string>();
+    const classes = useStyles();
+    const intl = useIntl();
+
+    const setIntlError = (id: string) => {
+        const message = intl.formatMessage({
+            id,
+        });
+        setError(message);
+    };
+
+    const onFileUpload = async (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e?.target?.files?.length) {
+            setUploading(true);
+            const file = e.target.files[0];
+            const text = await file.text();
+            try {
+                const keys = JSON.parse(text);
+                const context = {
+                    commitment_hash: keys.commitment_hash,
+                    crypto_base_hash: keys.crypto_base_hash,
+                    crypto_extended_base_hash: keys.crypto_extended_base_hash,
+                    elgamal_public_key: keys.elgamal_public_key,
+                } as CiphertextElectionContext;
+                const fileValid =
+                    context.commitment_hash &&
+                    context.crypto_base_hash &&
+                    context.crypto_extended_base_hash &&
+                    context.elgamal_public_key;
+                if (!fileValid) {
+                    throw new Error();
+                }
+                onChanged({ context } as SubmitElectionRequest);
+                onNext();
+            } catch (ex) {
+                setIntlError(MessageId.ElectionSetup_JointKeyUpload_InvalidFile);
+            } finally {
+                setUploading(false);
+            }
+        } else {
+            setIntlError(MessageId.ElectionSetup_JointKeyUpload_NoFile);
+        }
+    };
+
+    return (
+        <Box
+            display="flex"
+            flexDirection="column"
+            justifyItems="center"
+            alignItems="center"
+            className={classes.root}
+        >
+            <Container maxWidth="md" className={classes.content}>
+                <IconHeader
+                    title={new Message(MessageId.ElectionSetup_JointKeyUpload_Title)}
+                    Icon={KeyIcon}
+                />
+
+                <div className={classes.error}>{error}</div>
```

Comment:
If I was in angular I'd add `*ngIf="error"`, but I don't think react has any easy equivalent, so I guess this div will live here in the DOM and only have content if there's an error.